LLMonFHIR/FHIRInterpretation/UserStudy/UserStudyChatView.swift

```diff
@@ -53,13 +53,27 @@ struct UserStudyChatView: View {
                     ),
                     content: taskInstructionSheet
                 )
+                .viewStateAlert(state: viewModel.llmSession.state)
                 .onAppear(perform: viewModel.startSurvey)
                 .onChange(of: viewModel.llmSession.context, initial: true) {
                     Task {
                         if viewModel.isProcessing {
                             viewModel.updateProcessingState()
                         }
                         _ = await viewModel.generateAssistantResponse()
+                        
+                        // Alerts and sheets can not be displayed at the same time.
+                        if case let .error(error) = viewModel.llmSession.state {
+                            // We have to first dismiss all sheets.
+                            viewModel.setSurveyViewPresented(false)
+                            viewModel.setTaskInstructionSheetPresented(false)
+                            // Wait for animation to complete
+                            try await Task.sleep(for: .seconds(1))
+                            // Re-set the error state.
+                            viewModel.llmSession.state = .generating
+                            try await Task.sleep(for: .seconds(0.5))
+                            viewModel.llmSession.state = .error(error: error)
+                        }
```

Comment:
@jdisho Not perfect but addresses a typical SwiftUI pain point with conflicting presentation using sheets/alerts. Unfortunately SwiftUI doesn't resolve this internally and doesn't stack sheets/alerts. Probably hard to resolve in a general way. Timeout workaround is not great but works. We can consider increasing the sleep calls for older devices.

Reply:
So basically, we're catching errors thrown by SpeziLLM, waiting until the sheets are dismissed, and then propagating them back to SpeziLLM for presentation?